src/rf/apps/workers/process.py

```diff
@@ -0,0 +1,221 @@
+
+        s3_uuid = data['s3_uuid']
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+        image.status = STATUS_REPROJECTED
+        image.save()
+
+        data = {'layer_id': image.layer_id}
+        payload = self.make_payload(JOB_HANDOFF, data)
+        self.post_to_queue(payload)
+        return True
+
+    def emr_hand_off(self, data):
+        """
+        Passes layer imgages to EMR to begin creating custom rasters.
+        data -- attribute data from SQS.
+        """
+        # If all the images are ready send data to EMR for processing.
+        # return False if it fails to start.
+        # EMR posts directly to queue upon competion.
+
+        layer_id = data['layer_id']
+        layer_images = LayerImage.objects.filter(layer_id=layer_id)
+        reprojected_images = LayerImage.objects.filter(layer_id=layer_id,
+                                                       status=STATUS_REPROJECTED)  # NOQA
+        ready_to_process = len(layer_images) == len(reprojected_images)
+
+        if ready_to_process:
+            layer_meta = LayerMeta.objects.get(layer_id=layer_id)
+            layer_meta.state = STATUS_PROCESSING
+            layer_meta.save()
+
+            # POST TO EMR HERE.
+
+            # Add a message to the queue that we can use to watch for
+            # EMR timeout.
+            data = {
+                'timeout': time.time() + TIMEOUT_SECONDS,
+                'layer_id': layer_id
+            }
+
+            payload = self.make_payload(JOB_TIMEOUT, data)
+            self.post_to_queue(payload)
+
+        return True
+
+    def check_timeout(self, data):
+        """
+        Check an EMR job to see if it has completed before the timeout has
+        been reached.
+        data -- attribute data from SQS.
+        """
+
+        layer_id = data['layer_id']
+        timeout = data['timeout']
+        if time.time() > timeout:
+            layer_meta = LayerMeta.objects.get(layer_id=layer_id)
+            if not layer_meta.state == STATUS_COMPLETE:
+                layer_meta.state = STATUS_FAILED
+                layer_meta.save()
+            return True
+        else:
+            return False
+
+    def save_result(self, data):
+        """
+        When an EMR job has completed updates the associated model in the
+        database.
+        data -- attribute data from SQS.
+        """
+
+        # Update our server with data necessary for tile serving.
+
+        layer_id = data['layer_id']
+        layer_meta = LayerMeta.objects.get(id=layer_id)
+        layer_meta.state = STATUS_COMPLETE
+        layer_meta.save()
+        # Nothing else needs to go in the queue. We're done.
+        return True
+
+    def make_payload(self, type, data):
+        return {
+            'Records': [
+                {
+                    'eventSource': 'rf:boto',
+                    'eventName': type,
+                    'data': data
+                }
+            ]
+        }
```

Comment:
Don't think we necessarily need to follow the S3 data format. Also, `type` is shadowed here.